beta/components/header/index.js

```diff
@@ -276,19 +276,16 @@ class Header extends Component {
       `
     }
 
-    const renderDefault = () => {
-      return html`
-        ${this.renderSearch()}
-        ${nav()}
-      `
+    // Figure out which menu to render, only one can render at a time
+    let menu;
+    if (this.local.machine.state.library === 'on') {
+      menu = this.renderSubMenuItems({ name: 'library', eventName: 'library:toggle' }, this.local.machine)
+    } else if (this.local.machine.state.browse === 'on') {
+      menu = this.renderSubMenuItems({ name: 'browse', eventName: 'browse:toggle' }, this.local.machine)
+    } else {
+      menu = mainMenu()
     }
```

Comment:
Rather than using a complex dictionary/OR chain to figure out what to render in a single statement, I think using a normal control flow with if statements is easier to follow, at least for this statement. It reads more like how a human would think. Now you can easily see that when state X is Y, we render this specific menu, and when state Z is A, we render a different one, and so forth.